### PR TITLE
docs: add `ModifyColumn` on nightly

### PR DESCRIPTION
**Pull request by KKould**
## What's Changed in this PR

https://github.com/GreptimeTeam/greptimedb/issues/3517
https://github.com/GreptimeTeam/docs/issues/915

Add docs about new feature: "modify column type"

## Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

**Comment by nicecui** [comment]
@KKould Thank you very much for your contribution!
@MichaelScofield @evenyag please confirm the description and the db version of this feature. I consider it is in the db nightly version.

**Comment by evenyag** [comment]
> @KKould Thank you very much for your contribution! @MichaelScofield @evenyag please confirm the description and the db version of this feature. I consider it is in the db nightly version.

Yes, it should be the nightly version.

**Comment by nicecui** [comment]
@KKould Could you please move the content to the `docs/nightly` dictionary? Thank you very much

File: docs/nightly/en/reference/sql/alter.md [hunk 1]
```diff
@@ -54,6 +55,16 @@ ALTER TABLE monitor DROP COLUMN load_15;
 
 The removed column can't be retrieved immediately by all subsequent queries.
 
+### Modify column type
+
+Modify the date type of column
```
Comment (evenyag) [inline docs/nightly/en/reference/sql/alter.md:60]:
```suggestion
Modify the date type of a column
```

File: docs/nightly/en/reference/sql/alter.md [hunk 2]
```diff
@@ -54,6 +55,16 @@ ALTER TABLE monitor DROP COLUMN load_15;
...
+
+```sql
+ALTER TABLE monitor MODIFY COLUMN load_15 STRING;
+```
+
+The modified column cannot be a tag (primary key) or timestamp (time index), and it must be Nullable to ensure that the data can be safely converted (become null).
```
Comment (evenyag) [inline docs/nightly/en/reference/sql/alter.md:66]:
```suggestion
The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (become null).
```

File: docs/nightly/zh/reference/sql/alter.md [hunk 3]
```diff
@@ -55,6 +56,16 @@ ALTER TABLE monitor DROP COLUMN load_15;
...
 
+### 修改列类型
+
+修改列的数据类型
+
+```sql
+ALTER TABLE monitor MODIFY COLUMN load_15 STRING;
+```
+
+被修改的的列不能是 tag 列（primary key）或 time index 列，同时该列必须是 nullable 来保证数据能够安全地进行转换（变成 null）。
```
Comment (evenyag) [inline docs/nightly/zh/reference/sql/alter.md:67]:
```suggestion
被修改的的列不能是 tag 列（primary key）或 time index 列，同时该列必须是 nullable 来保证数据能够安全地进行转换（转换失败时返回 null）。
```

File: docs/nightly/en/reference/sql/alter.md [hunk 4]
```diff
@@ -54,6 +55,16 @@ ALTER TABLE monitor DROP COLUMN load_15;
...
 
+### Modify column type
+
+Modify the date type of a column
+
+```sql
+ALTER TABLE monitor MODIFY COLUMN load_15 STRING;
+```
+
+The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (become null).
```
Comment (evenyag) [inline docs/nightly/en/reference/sql/alter.md:66]:
```suggestion
The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (returns null on cast failures).
```

**Review by evenyag — APPROVED** [review]

**Review by nicecui — APPROVED** [review]

File: docs/nightly/zh/reference/sql/alter.md [hunk 5]
```diff
@@ -55,6 +56,16 @@ ALTER TABLE monitor DROP COLUMN load_15;
...
 
+### 修改列类型
+
+修改列的数据类型
+
+```sql
+ALTER TABLE monitor MODIFY COLUMN load_15 STRING;
+```
+
+被修改的的列不能是 tag 列（primary key）或 time index 列，同时该列必须是 nullable 来保证数据能够安全地进行转换（转换失败时返回 null）。
```
Comment (nicecui) [inline docs/nightly/zh/reference/sql/alter.md:67]:
```suggestion
被修改的的列不能是 tag 列（primary key）或 time index 列，同时该列必须允许空值 `NULL` 存在来保证数据能够安全地进行转换（转换失败时返回 `NULL`）。
```

File: docs/nightly/en/reference/sql/alter.md [hunk 6]
```diff
@@ -54,6 +55,16 @@ ALTER TABLE monitor DROP COLUMN load_15;
...
 
+### Modify column type
+
+Modify the date type of a column
+
+```sql
+ALTER TABLE monitor MODIFY COLUMN load_15 STRING;
+```
+
+The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (returns null on cast failures).
```
Comment (nicecui) [inline docs/nightly/en/reference/sql/alter.md:66]:
```suggestion
The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (returns `NULL` on cast failures).
```

**Pull request merged**